### ISSUE TITLE
DateAxisRangeSelector gets duplicated using useLayoutEffect for multiple props

### ISSUE THREAD

**Issue by RSchneider94** [amcharts/amcharts4#3088]
Hi!

As asked by @martynasma in [my other issue thread](https://github.com/amcharts/amcharts4/issues/3079) I'm opening this one with a reproducible example of the problem I'm having and can't see to make it work. The main problem that I thought that it was before, is that I couldn't have two different charts with `DateAxisRangeSelector`, because the second one was not working when clicking in the "quick filters"... But, after some research and tries I've found what was the main cause of the problem. I created a sandbox to show you what's happening:

https://codesandbox.io/s/amcharts-double-rangeselector-bug-n3r3v?file=/src/charts/FirstChart.jsx

@martynasma I copied your code example from the other thread to create the charts. Hope you don't mind 😬

The problem is with the `useLayoutEffect` dependencies `[chartId, data]`. Whenever the new data comes from the API, it will re-render and with that it will create a new `DateAxisRangeSelector`. Before, I was able to workaround this using React's `useRef` (it's still commented out in the code). But, then the click / change of periods will not work (my main problem) because the ref is from an old selector. So, following AmCharts' doc I saw that to handle dynamic data, we could do something like this instead of putting the dependencies in the main `useLayoutEffect`:

```javascript
useLayoutEffect(() => {
  chart.current.data = data;
}, [data]);
```

And that really works! You'll get just one `DateAxisRangeSelector` and the data will be correct! Cool! But... In my case, I have another param that comes dynamically from the API too and this param is something essencial that I use in the main `useLayoutEffect` to generate the chart. It's called `numSeries` and work to make all `LineSeries` that I need, here is the code:

```javascript
// Create series dynamically from data
for (let index = 0; index <= numSeries; index++) {
  const series = x.series.push(new am4charts.LineSeries());
  series.dataFields.dateX = 'date';
  series.dataFields.valueY = `price${index}Value`;
  series.tooltipText = `{price${index}Code}: R$ {valueY.value.formatNumber('#,###.00')}`;
  series.tooltip.getFillFromObject = false;
  series.tooltip.getStrokeFromObject = true;
  series.tooltip.background.fill = am4core.color('#1A1C1F');
  series.tooltip.background.strokeWidth = 2;
  series.tooltip.label.fill = series.stroke;
}
```

And the solution to just replace the data doesn't work because it will not re-render the chart. I really need to trigger a re-render whenever `numSeries` changes. So I found in the docs that `DateAxisRangeSelector` has a `.dispose()` method, like the chart, but it doesn't throw the control off and I still got two of them 😢 

Any ideas on how to fix this? Thanks!

**Comment by martynasma** [comment]
OK, so the issue is that RangeSelector is not being properly disposed.

We'll fix that in the next version (due to be released very soon).

After update RangeSelector will auto-dispose itself when related axis is disposed OR when you call `dispose()` on it directly.

I'll notify you here when new version is out.

**Comment by RSchneider94** [comment]
> OK, so the issue is that RangeSelector is not being properly disposed.
> 
> We'll fix that in the next version (due to be released very soon).
> 
> After update RangeSelector will auto-dispose itself when related axis is disposed OR when you call `dispose()` on it directly.
> 
> I'll notify you here when new version is out.

hi @martynasma !! I see, understood! Thank you! 😊

**Comment by martynasma** [comment]
Fixed in [4.10.11](https://github.com/amcharts/amcharts4/releases/tag/4.10.11).

## [4.10.11] - 2020-11-13

### Added
- New setting `reverseGeodata` (default: `false`) on `MapChart` and `MapSeries`. amCharts requires polygon coordinates to be in clockwise order. Some map data have them in counter-clockwise. If your custom map appears garbled, try setting this to `true`.
- New global option: `am4core.options.pixelPerfectPrecision` (default: `0`). Precision (number of decimal points) to be used when rounding point x/y for elements that have `pixelPerfect = true` set.

### Fixed
- `XYChartScrollbar` was not inheriting `inputDateFormat` from the chart properly.
- Clicking on a chart on mobile devices could sometimes make them document scroll a bit.
- Accessibility: pressing ENTER while there is a focused element with `showTooltipOn = "hit"` will not display its tooltip.
- RangeSelector (plugin) was not properly removing its elements after `dispose()` it will now auto-dispose when related axis is disposed.
- In some cases a console error could be displayed when using `XYCursor` over chart when data was being updated.
- Very long labels with `wrap = true` may have been wrapping incorrectly in some cases.
- JSON config: `rgb(...)` and `rgba(...)` syntax was not recongnised in `ColorSet`'s `baseColor`.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
